File: maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int maxTot = nums[0];
        int maxCurr = nums[0];
        
        for(int i = 1; i<n; i++){
            maxCurr = max(nums[i],nums[i]+maxCurr);
            if(maxCurr > maxTot){
                maxTot = max(maxTot,maxCurr);
            }
        }
        
        return maxTot;
    }
};

File: maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int maxTillNow = nums[0];
        
        for(int i = 1; i < nums.length; i++){
            maxTillNow = Math.max(nums[i], maxTillNow+nums[i]);
            if(maxTillNow > maxSum) {
                maxSum = maxTillNow;
            }
        }
        
        return maxSum;
    }
}